src/main/java/com/flansmod/common/guns/EntityGrenade.java

```diff
@@ -563,7 +624,18 @@ public void writeSpawnData(ByteBuf data)
 	public void readSpawnData(ByteBuf data)
 	{
 		type = GrenadeType.getGrenade(ByteBufUtils.readUTF8String(data));
-		thrower = (EntityLivingBase)world.getEntityByID(data.readInt());
+		{
+			Entity ent = world.getEntityByID(data.readInt());
+			if (ent instanceof EntityPlayer)
```

Comment:
`player = ent instanceof EntityPlayer ? Optional.of((EntityPlayer) ent) : Optional.empty();` preferred, especially with this code formatting where braces are on new lines :roll_eyes: